### ISSUE TITLE
exif_data_load_data fails to load exif from WebP

### ISSUE THREAD

**Issue by mnaydenov** [libexif/libexif#58]
The title says it all. The reason is, `exif_data_load_data` assumes `APP1` marker data, more specifically it insist, the data starts with "exif\0\0". 
In the case of WebP (and probably others, like PNG), where exif is stored in a chunk, the data starts right with the TIFF Header (byte order, followed by 42 and 0th IDF Offset).

P.S. To test and/or verify one can convert a JPEG to a WebP, using ImageMagic `convert` or GIMP Export As...

**Comment by msmeissn** [comment]
yeah, definitely seems missing. lets see if we can find and add a file reader for webp

**Comment by romanb089** [comment]
> yeah, definitely seems missing. lets see if we can find and add a file reader for webp

Please tell me if there is any news on this issue?

**Comment by msmeissn** [comment]
i did have a short look at our file reader, but its kind of complicated and I am not familar with it yet.
so I did not get to it yet

**Comment by mnaydenov** [comment]
Hello again, observed AVIF and HEIF files with the same behavior. 
What is more the, the ones that do have "exif\0\0" at the start also have non zero offset of 6, effectively instructing the reader to skip over the signature. 
![image](https://user-images.githubusercontent.com/13903975/133887019-5852d3a8-5f40-40a1-999f-3b6d5c56e80a.png)
![image](https://user-images.githubusercontent.com/13903975/133887024-b589aa7d-4628-4185-a958-8918c7e8863d.png)

**Comment by msmeissn** [comment]
yes, we need other file importers.

i spent some time with enhancing other parts of libexif so far, e.g. add more makernote names for Nikon and Canon.
this is something we can pick up

**Comment by mnaydenov** [comment]
I personally use libexif by giving it the exif as a blob. Image loading libraries seem to be good at getting (and setting) this data as a blob, but editing it is though or out of scope and this is where libexif comes in.
Right now my workaround is simply to prepend "Exif\0\0" to the blob and all is fine.
